templates/utility.html

```diff
@@ -13,11 +12,22 @@
         {{error}}
     </div>
     {% endif %}
-
     {% set vars = {"readonly": "readonly" if status == "check" else "", "bank_index": 0} %}
 
     <div class="card">
-        <div class="card-header">
+    {% if not kyc_info['level'] < '2' %}
```

Comment:
I think the oauth html stuff should go into its own template (like oauth.html)

Reply:
can't find a way of doing this. investigating.

Reply:
can't find a way to make sure that the utility pages are inaccessible without the following:
```
                else:
                    error = "failed to create invoice ({})".format(err)
        return render_template("utility.html", utility=utility, selected_bank_name=bank_desc["name"], status=status, email=email, amount=amount, utility_name=utility_name, values=values, error=error, kyc_info=kyc_info, userinfo=userinfo_content, url=url)
    else:
        return render_template("utility.html", utility=utility, status=STATUS_CREATE, values=werkzeug.MultiDict(), kyc_info=kyc_info, userinfo=userinfo_content)
```

so what ive done is using "abort" to error the page if its not "ok".

Reply:
changed this again.
```
    kyc = bronze_blueprint.session.get('AccountKyc')
    if not kyc.ok:
        flash('Issue retrieving KYC information', 'error')
        return redirect(url_for('oauth'))
    kyc_info = kyc.json()
    level = kyc_info['level']
    userinfo = bronze_blueprint.session.get('UserInfo')
    if not userinfo.ok:
        flash('Issue retrieving UserInfo information', 'error')
        return redirect(url_for('oauth'))
    userinfo_content = userinfo.json()
```
so if its not ok, it would redirect to route 'oauth' this is then redirected to the 'utilities' page since im already authorised where it will show if there is a problem getting the KYC info or the USERINFO, which is needed to render the  'utility' template. Inside the utility template there is the check that asses whether the kyc level is high enough or not and if not it show show the button to login to the xchange the verify the kyc part. If this needs to happen, there;s no way to move the logic to oauth because you are getting redirected automatically to somewhere else after the oauth.